pkg/api/validation/validation.go

```diff
@@ -38,8 +38,7 @@ func validateVolumes(volumes []api.Volume) (util.StringSet, errs.ValidationError
 	allErrs := errs.ValidationErrorList{}
 
 	allNames := util.StringSet{}
-	for i := range volumes {
-		vol := &volumes[i] // so we can set default values
+	for i, vol := range volumes {
```

Comment:
Our encoding / decoding / conversion is an order of magnitude worse (we do 2000 allocations per decode, and actually encode 3 times).  A few copies we can probably let slide...


> On Jan 29, 2015, at 7:31 PM, Tim Hockin <notifications@github.com> wrote:
> 
> In pkg/api/validation/validation.go:
> 
> > @@ -38,8 +38,7 @@ func validateVolumes(volumes []api.Volume) (util.StringSet, errs.ValidationError
> >  	allErrs := errs.ValidationErrorList{}
> >  
> >  	allNames := util.StringSet{}
> > -	for i := range volumes {
> > -		vol := &volumes[i] // so we can set default values
> > +	for i, vol := range volumes {
> This is a good way to prove that we are not depending on mutating changes, but it is pessimal for performance - it will make a copy of every object. I think we should stick with pointers.
> 
> —
> Reply to this email directly or view it on GitHub.
>